packages/shared/lib/services/nango-config.service.ts

```diff
@@ -497,43 +496,43 @@ export function getOffset(interval: string, date: Date): number {
  * and then 1636 etc. The offset should be based on the interval and should never be
  * greater than the interval
  */
-export function getInterval(runs: string, date: Date): ServiceResponse<{ interval: string; offset: number }> {
+export function getInterval(runs: string, date: Date): ServiceResponse<{ interval: StringValue; offset: number }> {
     if (runs === 'every half day') {
-        const response = { interval: '12h', offset: getOffset('12h', date) };
+        const response = { interval: '12h' as StringValue, offset: getOffset('12h', date) };
```

Comment:
I would create a type instead of using `as` (or return directly); this keyword will cast the string even tho it's not compatible. e.g: `'foobar' as StringValue` will work but should not

```ts
interface GetIntervalRes { interval: StringValue; offset: number }
export function getInterval(runs: string, date: Date): ServiceResponse<GetIntervalRes> {
   const response: GetIntervalRes = { interval: '12h', offset: getOffset('12h', date) };
   return { success: true, error: null, response };
}
```
 
alternative with type inference

```ts
export function getInterval(runs: string, date: Date): ServiceResponse<{ interval: StringValue; offset: number }> {
   return { success: true, error: null, response: { interval: '12h', offset: getOffset('12h', date) } };
}
```

Reply:
Thanks, updated.